src/components/TextField/TextField.tsx

```diff
@@ -0,0 +1,247 @@
+
+    if (inputElem.value.length <= 0) inputElem.placeholder = placeholder || '';
+
+    onBlur(e);
+    updateStatusFromInputValue(inputElem.value);
+  };
+
+  const handleFocus: FocusEventHandler<HTMLInputElement> = () => {
+    const inputElem = getInitializedRef(inputRef);
+
+    if (inputElem.value === '') inputElem.placeholder = '';
+
+    updateStatusFromInputValue(inputElem.value);
+  };
+
+  const handleInputChange: ChangeEventHandler<HTMLInputElement> = async e => {
+    const inputElem = getInitializedRef(inputRef);
+    const lengthCountElem = getInitializedRef(lengthCountRef);
+
+    lengthCountElem.innerText = getStringOfValueLengthPerMax(
+      inputElem.value,
+      max
+    );
+
+    onChange(e);
+    const valdationResult = await validate(inputElem.value);
+
+    if (valdationResult.result === true) return setTextFieldStatus('active');
+    if (valdationResult.type === 'max')
+      return setTextFieldStatus('error', '글자수를 초과했습니다.');
+    if (valdationResult.type === 'email')
+      return setTextFieldStatus('error', '이메일 형식이 아닙니다.');
+  };
+
+  return (
+    <div
+      className={clsx([style.inputTypeRecipe({ status }), style.wrapper])}
+      arial-lable="text"
+    >
+      <label className={clsx([style.label, variants.caption1])}>{label}</label>
+      <div className={style.inputContainer}>
+        <input
+          name={name}
+          type={type}
+          className={style.input({ size })}
+          placeholder={placeholder}
+          onChange={handleInputChange}
+          onBlur={handleBlur}
+          onFocus={handleFocus}
+          ref={node => {
+            inputRef.current = node; // TextField 내부의 inputRef 사용
+            if (typeof ref === 'function') ref(node); // RefCallback이 오는 경우 적용
+          }}
+        />
+        <RemoveButton
+          onClick={handleRemoveClick}
+          inputRef={inputRef}
+          status={status}
+        />
+        <Count max={max} ref={lengthCountRef} />
+        <div className={style.underbar} />
+      </div>
+      <Message status={status} message={message} />
+    </div>
+  );
+});
+export default TextField;
+
+/**
+ *
+ */
+const RemoveButton = ({
+  onClick,
+  inputRef
+}: {
+  onClick: MouseEventHandler<HTMLButtonElement>;
+  inputRef: { current: HTMLInputElement | null };
+  status: TextFieldStatus;
+}) => {
+  const checkIsOnOff = () =>
+    inputRef.current && inputRef.current.value.length && status !== 'default'
+      ? 'on'
+      : 'off';
+
+  return (
+    <button onClick={onClick}>
+      <TextFieldRemoveIcon
+        className={clsx([style.icon({ status: checkIsOnOff() })])}
+      />
+    </button>
+  );
+};
+
+/**
+ * 글자 수 카운트하는 컴포넌트
+ */
+const Count = React.forwardRef(function Count(
+  { max }: { max?: number | string },
+  ref: ForwardedRef<HTMLDivElement>
+) {
+  return (
+    <>
+      <div
+        ref={ref}
+        className={`${max && clsx([style.count, variants.body3])}`}
+      >
+        {max && `0/${max}`}
+      </div>
```

Comment:
typography 컴포넌트(`<Body3/>`) 대신 typography variants를 임포트해서 쓰시는 이유가 있으실까요??

```suggestion
      <Body3
        ref={ref}
        className={clsx({[style.count]: Boolean(max)})}
      >
        {max && `0/${max}`}
      </div>
```